src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/Fast5FailFolder.groovy

```diff
@@ -3,33 +3,33 @@ package life.qbic.datamodel.datasets.datastructure.folders.nanopore
 import life.qbic.datamodel.datasets.datastructure.folders.DataFolder
 
 /**
- * A special case of a DataFolder, its name is always fast5_fail.
+ * A special case of a DataFolder, its name contains always fast5_fail.
  *
  * Its children field contains either a list of type List<Fast5Files> or List<Fast5Folder>
  *
  * @author: Sven Fillinger
  */
 class Fast5FailFolder extends DataFolder {
 
-    final private static String NAME = "fast5_fail"
-
     final private static String NAME_SCHEMA = /fast5_fail/
 
     protected Fast5FailFolder() {}
 
-    protected Fast5FailFolder(String relativePath, List children) {
-        super(NAME, relativePath, children)
+    protected Fast5FailFolder(String name, String relativePath, List children) {
+        super(name, relativePath, children)
```

Comment:
When is the given name validated? Should this be done before creating the object?

Reply:
It is part of it. But you must always call `super()` before you can continue if you use inheritance. The validation throws a runtime exception upon business rule violation.